### PR TITLE
Add RabbitMQ single active consumer argument

### PR DESCRIPTION
**Pull request by DropSnorz**
# Description

Added a new RabbitMQ subscription argument to enable [Single Active Consumer](https://www.rabbitmq.com/docs/consumers#single-active-consumer) (`x-single-active-consumer`)

Usage with declarative subscription
```yaml
apiVersion: dapr.io/v2alpha1
kind: Subscription
metadata:
  name: orders
spec:
  topic: orders
  routes:
    default: /orders
  pubsubname: orderpubsub
  metadata:
    queueName: "foo-queue-quorum"
    queueType: quorum
    singleActiveConsumer: "true"
scopes:
- order-processor-sdk

```

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #3120

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

File: pubsub/rabbitmq/rabbitmq_test.go [hunk 1]
```diff
@@ -129,6 +129,11 @@ func TestPublishAndSubscribeWithPriorityQueue(t *testing.T) {
 	<-processed
 	assert.Equal(t, 4, messageCount)
 	assert.Equal(t, "foo bar", lastMessage)
+
+	// subscribe using single active consumer
+	err = pubsubRabbitMQ.Subscribe(context.Background(), pubsub.SubscribeRequest{Topic: topic, Metadata: map[string]string{reqMetadataSingleActiveConsumerKey: "true"}}, handler)
```
Comment (berndverst) [inline pubsub/rabbitmq/rabbitmq_test.go:134]: You need to actually publish a message after this, then assert that the message was received and the overall messageCount has been increased.

File: pubsub/rabbitmq/rabbitmq.go [hunk 2]
```diff
@@ -432,6 +434,18 @@ func (r *rabbitMQ) prepareSubscription(channel rabbitMQChannelBroker, req pubsub
 		args[amqp.QueueTypeArg] = amqp.QueueTypeClassic
 	}
 
+	// Applying x-single-active-consumer if defined at subscription level
+	if val := req.Metadata[reqMetadataSingleActiveConsumerKey]; val != "" {
+		parsedVal, pErr := strconv.ParseBool(val)
```
Comment (berndverst) [inline pubsub/rabbitmq/rabbitmq.go:439]: Do not use `strconv.ParseBool` instead use our preferred method from package `dapr/kit/utils` called `IsTruthy` https://github.com/dapr/kit/blob/106329e5839f70f3234b9f49b7dabff0247c0a90/utils/strings.go#L23

That will allow a lot more values to be used to express the "true" / on state.

**Review by berndverst — CHANGES_REQUESTED** [review]

**Comment by berndverst** [comment]
A few small changes / additions, then we can merge this.

File: pubsub/rabbitmq/rabbitmq.go (hunk 2 above)
Comment (DropSnorz) [reply pubsub/rabbitmq/rabbitmq.go]: Thanks for highlighting this, the code is far more readable this way 👍

File: pubsub/rabbitmq/rabbitmq_test.go (hunk 1 above)
Comment (DropSnorz) [reply pubsub/rabbitmq/rabbitmq_test.go:134]: Test function updated.

**Review by yaron2 — APPROVED** [review]

**Review by yaron2 — APPROVED** [review]

**Comment by yaron2** [comment]
@DropSnorz Thanks for this PR! Please open a docs issue in dapr/docs to update the RabbitMQ pub/sub spec with this setting.

File: pubsub/rabbitmq/rabbitmq.go [hunk 3]
```diff
@@ -48,17 +49,19 @@ const (
...
-	reqMetadataRoutingKey     = "routingKey"
-	reqMetadataQueueTypeKey   = "queueType" // at the moment, only supporting classic and quorum queues
-	reqMetadataMaxLenKey      = "maxLen"
-	reqMetadataMaxLenBytesKey = "maxLenBytes"
+	argQueueMode                       = "x-queue-mode"
+	argMaxLength                       = "x-max-length"
+	argMaxLengthBytes                  = "x-max-length-bytes"
+	argDeadLetterExchange              = "x-dead-letter-exchange"
+	argMaxPriority                     = "x-max-priority"
+	argSingleActiveConsumer            = "x-single-active-consumer"
```
Comment (ItalyPaleAle) [inline pubsub/rabbitmq/rabbitmq.go:57]: Please open another PR to update https://github.com/dapr/components-contrib/blob/main/pubsub/rabbitmq/metadata.yaml
CC: @yaron2

Comment (DropSnorz) [reply pubsub/rabbitmq/rabbitmq.go:57]: @ItalyPaleAle It's not clear to me what should be updated in the `metadata.yml` file. This file seems to describe metadata for the rabbitmq pubsub **Component** while the `singleActiveConsumer` parameter is set at the **Subscription** level (like `queueName`, `routingKey`, ...)

Comment (ItalyPaleAle) [reply pubsub/rabbitmq/rabbitmq.go:57]: I see, I misunderstood this PR, thanks for clarifying.

**Comment by marcduiker** [comment]
@holopin-bot @DropSnorz Thank you!

**Pull request merged**